### ISSUE TITLE
[featrure] Fatjar support

### ISSUE THREAD

**Issue by TobseF** [AlexisJehan/MvnCheck#5]
The application layout with `bin`, `conf` and `lib` is good for developing or using it inside Docker. But I would prefer to have the option for a fat-jar with all dependencies. This makes it easier to handle to just need to download a single file. And execution is simple because I do not need to specify the classpath.

My final goal is to compile it to a native image with GraalVM, so distribution is also independent from the jvm.

I already played around with the config of the `maven-assembly-plugin`.
I added the `jar-with-dependencies` to the `configuration` but was not able to get it working:
```xml
<descriptorRefs>
    <descriptorRef>jar-with-dependencies</descriptorRef>
</descriptorRefs>
<archive>
<manifest>
    <mainClass>com.github.alexisjehan.mvncheck.Application</mainClass>
</manifest>
</archive>
```

Of course, I can understand that this is not a must-have feature, and It may be additional work to support two assembly modes.

**Comment by AlexisJehan** [comment]
Hello,
Even if I didn't thought about it, of course this is a great idea.

I had a look, and I identified two problems with it:
- The _Maven Resolver_ implementation is based on _Eclipse Sisu_ (which is based on _Google Guice_) and that doesn't work well inside the JAR with dependencies.
- The `init.gradle` file from the classpath couldn't be passed as an argument anymore to _Gradle_.

Solutions:
- Migrate from _Eclipse Sisu_ to the new _Supplier_ implementation of _Maven Resolver_ (will be compliant with the version 2 in the future)
- Write the content of the `init.gradle` file to a temporary file that can be passed to _Gradle_

**Comment by AlexisJehan** [comment]
Please give it a try and feel free to report me any feedback in case you encounter some issue with it!

**Comment by AlexisJehan** [comment]
Released in [1.7.0](https://github.com/AlexisJehan/MvnCheck/releases/tag/v1.7.0).
